src/domains/chart/sagas.ts

```diff
@@ -123,6 +126,13 @@ function* fetchDataSaga({ payload }: Action<FetchDataPayload>) {
     ? `${alwaysEndWithSlash(host)}api/v1/data`
     : host
 
+  const agentOptionsOriginal = options.split("|")
+  const hasFlip = agentOptionsOriginal.includes("flip")
+  const shouldAddFakeFlip = !IS_FLIP_RESPECTED_IN_COMPOSITE_CHARTS && !hasFlip
+  // if flip is not respected in composite-charts, send it always (like dygraph charts normally do)
+  const agentOptions = shouldAddFakeFlip
+    ? agentOptionsOriginal.concat("flip") : agentOptionsOriginal
+
```

Comment:
Since the `split, includes` are executed once the flag `IS_FLIP_RESPECTED_IN_COMPOSITE_CHARTS` is true, i suggest change the code in order to execute the extra code only when it is required. Maybe something like `transformOprtions(options) {if (IS_FLIP_RESPECTED_IN_COMPOSITE_CHARTS) return options.split("|") else ...}`

Reply:
`split` needs to be done in all conditions actually. So it's just the `includes` check, don't you think it's over-optimisation since this is code is only running when doing /data calls - and `includes` check takes only fraction of a ms?
My intention was to have the flag calculated once so it can be later used when transforming the response.